### PR TITLE
Fix sporadic failures in nestedOn and remote-array-write-access

### PR DESCRIPTION
**Pull request by vasslitvinov**
This PR fixes sporadic failures in these two tests, as suggested by @ronawho.

### test/performance/ferguson/remote-array-write-access.chpl

This should fix the sporadic failure in `performance/ferguson/remote-array-write-access.chpl` when compiled with `--cache-remote` enabled.

The failure mode, most recently observed in our `gasnet.darwin-m1` configuration, is:
```
ERROR -- TOO MANY GETs, 2 EXCEEDS MAX OF 1
```

When Locale 1 issues a cache-get of A._instance, which is a wide pointer, it is typically done with a single get_nb. However, sometimes the _instance field is split across two cache pages, so one cache-get results in two get_nb communications. This PR allows this test to accept either 1 or 2 get_nb.

We limit this adjustment to this configuration; everywhere else it will continue requiring 1 get_nb. This way we will be alerted to extra comms in other configurations if/when they arise in the future.

### test/optimizations/remoteValueForwarding/gmp/nestedOn.chpl

This test has nested on-stmts and the inner one halts. We sporadically get warnings like: `GASNET WARNING(Node 0): int AMUDP_DrainNetwork(ep_t) returning an error code: AM_ERR_RESOURCE` for this test.

Just add a prediff to filter out any extra messages and only look for our expected error.

Once the .future is resolved, we will be getting mismatches on both .bad and .good. At this point the prediff needs to be removed.

This is analogous to https://github.com/chapel-lang/chapel/pull/12937 and https://github.com/chapel-lang/chapel/pull/14837, which added these prediffs:

    test/multilocale/bradc/needMultiLocales/writeThisUsingOn.prediff
    test/multilocale/local/diten/test_local2.prediff

### CODE REVIEW

**Comment by ronawho** [comment]
See the latest comments on https://github.com/Cray/chapel-private/issues/3728. I don't think the extra GET is from a readahead and even if it was subtracting the number of cache misses isn't appropriate since that's not the number of headaheads.

File: test/performance/ferguson/remote-array-write-access.chpl [hunk 1]
```diff
@@ -17,4 +17,4 @@ stop();
 writeln(A[1]);
 writeln(A[n]);
 
-report(maxGets=1, maxOns=1);
+report(maxGets=2, maxOns=1, maxGetsToAdjust=1);
```
Comment (ronawho) [inline test/performance/ferguson/remote-array-write-access.chpl:20]: I don't follow the `maxGetsToAdjust` logic (not saying it's wrong, I just don't grok it). Left to my own devices, I would just change this test to do something like:

```chpl
var maxGets = 1;
// accept higher comm counts on m1 mac for now -- see https://github.com/Cray/chapel-private/issues/3728
if CHPL_TARGET_PLATFORM == "darwin" && CHPL_TARGET_ARCH == "arm64" then maxGets = 2;
report(maxGets=maxGets, maxOns=1, maxGetsToAdjust=1);
```

to make it more obvious that it's just for arm-based macs we're seeing this, but maybe there's a benefit of the `maxGetsToAdjust` approach I don't understand.

**Comment by vasslitvinov** [comment]
Good question. `maxGetsToAdjust` is there to avoid "upper bound comm count is not tight in wost-case configuration" that I'd get when there is no split across cache lines and compopts include --checkMaxAttained. I can clarify further.

You are proposing to do the adjustment only for the particular configuration. Which will allow us to notice the cache issue in another configuration in the future. I am game. Combining your change with my change, the adjustment would apply to maxGetsToAdjust instead of maxGets:
```chpl
var maxGetsToAdjust = 1;
// accept higher comm counts on m1 mac for now -- see https://github.com/Cray/chapel-private/issues/3728
if CHPL_TARGET_PLATFORM == "darwin" && CHPL_TARGET_ARCH == "arm64" then maxGetsToAdjust = 2;
report(maxGets=1, maxOns=1, maxGetsToAdjust=maxGetsToAdjust);
```

How does this sound?

**Comment by vasslitvinov** [comment]
Scratch that. The adjustment for m1 needs to be done to maxGets as you proposed, not maxGetsToAdjust.

@ronawho I pushed the change. Good to go?

The test passes on chapcs (1 trial). I am testing it on the mac now -- 100 trials before and after this PR. I will verify that there are before and no failures after.

Thanks again.

**Review by ronawho — APPROVED** [review]

**Pull request merged**